crates/cli-support/src/js/js2rust.rs

```diff
@@ -192,7 +192,74 @@ impl<'a, 'b> Js2Rust<'a, 'b> {
         }
 
         if optional {
-            bail!("unsupported optional argument to rust function {:?}", arg);
+            if arg.is_primitive() {
+                self.cx.expose_is_like_none();
+                self.js_arguments.push((name.clone(), "number".to_string()));
+
+                if self.cx.config.debug {
+                    self.cx.expose_assert_num();
+                    self.prelude(&format!(
+                        "\n\
+                            if (!isLikeNone({0})) {{\n\
+                                _assertNum({0});\n\
+                            }}\n\
```

Comment:
Nowadays all the `\n\` markers here should be fine to clean up, the auto-formatting will strip them all away anyway!